src/utils/work/downloadFromS3.js

```diff
@@ -0,0 +1,29 @@
+import unpackResult from 'utils/work/unpackResult';
+import httpStatusCodes from 'utils/http/httpStatusCodes';
+import parseResult from 'utils/work/parseResult';
+
+const downloadFromS3 = async (taskName, signedUrl) => {
+  const response = await fetch(signedUrl);
+
+  // some WorkRequests like scType and runClustering do not upload data to S3
+  // (nor return it via socket) instead they patch the cellsets through the API.
+  // In those cases the workResults will not exist, and it's fine because data will
+  // be updated through the cellsets.
+  // the forbidden (in addition to the not found) is required because of how signed URLs work
+  //  when you try to download a file from a signedUrl that doesn't exist you get a 403 forbidden
+  // because the user is not authorized to access a file that does not exist
```

Comment:
This is a nice comment, but could we not make these cases (runClustering, etc) not even call `downloadFromS3`?

Reply:
I prefer this, it is a single code path robust to failures or situations where it does not exist. THe other option is to keep adding "ifs" to the code which I don't like. The better solution would be to stop sending that `signedUrl` from the worker after your refactor. But it's not great because as it is, it would require adding a new parameter to all worker task types